### PR TITLE
Add clang format from hdf5

### PR DESCRIPTION
**Pull request by glennsong09**

### CODE REVIEW

**Comment by glennsong09** [comment]
The format source script has problems when we brought it over, we will fix it later.

File: vol_test_parallel.c [hunk 1]
```diff
@@ -42,54 +42,54 @@ uint64_t vol_cap_flags;
...
+    X(VOL_TEST_NULL, "", NULL, 0)                                                                            \
+    X(VOL_TEST_FILE, "file", vol_file_test_parallel, 1)                                                      \
+    X(VOL_TEST_GROUP, "group", vol_group_test_parallel, 1)                                                   \
+    X(VOL_TEST_DATASET, "dataset", vol_dataset_test_parallel, 1)                                             \
+    X(VOL_TEST_DATATYPE, "datatype", vol_datatype_test_parallel, 1)                                          \
+    X(VOL_TEST_ATTRIBUTE, "attribute", vol_attribute_test_parallel, 1)                                       \
+    X(VOL_TEST_LINK, "link", vol_link_test_parallel, 1)                                                      \
+    X(VOL_TEST_OBJECT, "object", vol_object_test_parallel, 1)                                                \
+    X(VOL_TEST_MISC, "misc", vol_misc_test_parallel, 1)                                                      \
+    X(VOL_TEST_MAX, "", NULL, 0)
```
Comment (derobins) [inline vol_test_parallel.c:68]: Not sure how important it would be to turn the formatter off here

File: vol_misc_test.c [hunk 2]
```diff
@@ -23,31 +23,29 @@ static int test_double_init_term(void);
...
 static int (*misc_tests[])(void) = {
-        test_open_link_without_leading_slash,
-        test_object_creation_by_absolute_path,
-        test_absolute_vs_relative_path,
-        test_dot_for_object_name,
-        test_symbols_in_compound_field_name,
-        test_double_init_term,
+    test_open_link_without_leading_slash, test_object_creation_by_absolute_path,
+    test_absolute_vs_relative_path,       test_dot_for_object_name,
+    test_symbols_in_compound_field_name,  test_double_init_term,
```
Comment (derobins) [inline vol_misc_test.c:28]: Uglier

File: vol_test_parallel.c (hunk 1 above)
Comment (jhendersonHDF) [reply vol_test_parallel.c:68]: This one doesn't look too important. I'm guessing it's just trying to pad out spaces to match the line length, but we could maybe check clang format's options to see what it's trying to do here.

File: vol_misc_test.c (hunk 2 above)
Comment (jhendersonHDF) [reply vol_misc_test.c:28]: Yeah I think this is definitely one annoying habit of clang-format. Sure, if you have a lot of tests it'll start to take up vertical space, but having two on a line makes it harder to find what you're looking for.

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**